src/transfer.jl

```diff
@@ -203,25 +200,65 @@ function find_equivalent_kpt(basis::PlaneWaveBasis{T}, kcoord, spin; tol=sqrt(ep
     return (; index, ΔG)
 end
 
-"""
-Return the Fourier coefficients for `ψk · e^{i q·r}` in the basis of `kpt_out`, where `ψk`
-is defined on a basis `kpt_in`.
-"""
-function multiply_by_expiqr(basis, kpt_in, q, ψk)
-    shifted_kcoord = kpt_in.coordinate .+ q  # coordinate of ``k``-point in ℝ
-    index, ΔG = find_equivalent_kpt(basis, shifted_kcoord, kpt_in.spin)
-    kpt_out = basis.kpoints[index]
-    return transfer_blochwave_kpt(ψk, basis, kpt_in, kpt_out, ΔG)
-end
-
 """
 Return the indices of the `kpoints` shifted by `q`. That is for each `kpoint` of the `basis`:
-`kpoints[ik].coordinate + q = kpoints[indices[ik]].coordinate`.
+`kpoints[ik].coordinate + q` is equivalent to `kpoints[indices[ik]].coordinate`.
 """
-function k_to_kpq_mapping(basis::PlaneWaveBasis, qcoord)
+function k_to_equivalent_kpq_permutation(basis::PlaneWaveBasis, qcoord)
     kpoints = basis.kpoints
     indices = [find_equivalent_kpt(basis, kpt.coordinate + qcoord, kpt.spin).index
                for kpt in kpoints]
-    @assert sort(indices) == 1:length(basis.kpoints)
+    @assert isperm(indices)
     indices
 end
+
+@doc raw"""
+Create the Fourier expansion of ``ψ_{k+q}`` from ``ψ_{[k+q]}``, where ``[k+q]`` is in
+`basis.kpoints`. while ``k+q`` may or may not be inside.
+
+If ``ΔG ≔ [k+q] - (k+q)``, then we have that
+```math
+    ∑_G \hat{u}_{[k+q]}(G) e^{i(k+q+G)·r} &= ∑_{G'} \hat{u}_{k+q}(G'-ΔG) e^{i(k+q+ΔG+G')·r},
+```
+hence
+```math
+    u_{k+q}(G) = u_{[k+q]}(G + ΔG).
+```
+"""
+function kpq_equivalent_blochwave_to_kpq(basis, kpt, q, ψk_plus_q_equivalent)
+    kcoord_plus_q = kpt.coordinate .+ q
+    index, ΔG = find_equivalent_kpt(basis, kcoord_plus_q, kpt.spin)
+    equivalent_kpt_plus_q = basis.kpoints[index]
+
+    kpt_plus_q = build_kpoints(basis, [kcoord_plus_q])[kpt.spin]
+    (; kpt=kpt_plus_q,
+     ψk=transfer_blochwave_kpt(ψk_plus_q_equivalent, basis, equivalent_kpt_plus_q,
+                               kpt_plus_q, -ΔG))
+end
+
+# Replaces `multiply_by_expiqr`.
+# TODO: Think about a clear and consistent way to define such a function when completing
+# phonon PRs.
+@doc raw"""
+    multiply_ψ_by_blochwave(basis::PlaneWaveBasis, ψ, f_real, q)
+
+Return the Fourier coefficients for the Bloch waves ``f^{\rm real}_{q} ψ_{k-q}`` in an
+element of `basis.kpoints` equivalent to ``k-q``.
+"""
+function multiply_ψ_by_blochwave(basis::PlaneWaveBasis, ψ, f_real, q)
```

Comment:
Yeah finding good names for these functions is tricky. I'm not sure we are there yet. So perhaps it's good to document extra clearly why they exist, e.g. I think would be good if you mention the nice explanation δHψ_k = δV_{q} · ψ_{k-q}. here.